Add option to connect to plex server over https

tenfire opened the pull request:
Solves https://github.com/music-assistant/hass-music-assistant/issues/1968

Manual configuration is needed for https/ssl to work, since it is still hardcoded to use http for plex gdm discovery.

marcelveldt commented:
@micha91 and/or @jozefKruszynski I leave this one for you to review as you have more insight about Plex.
To me the Plex setup is getting rather daunting with all these options now

marcelveldt commented on music_assistant/server/providers/plex/__init__.py:
```diff
@@ -4,6 +4,7 @@
 
 import asyncio
 import logging
+import requests
```
verify if requests is part of the plex library, otherwise this import will fail.
we are not doing anything with requests but its async sibbling, aiohttp.

tenfire replied:
I'm not sure how to check this.
Found this at least: https://github.com/pkkid/python-plexapi/blob/abcab4f78d22e86b380327c6014d3148e0248e6d/plexapi/server.py#L7

marcelveldt replied:
Python plexapi has requests as its (sole) requirement, so the answer is yes.
https://github.com/pkkid/python-plexapi/blob/abcab4f78d22e86b380327c6014d3148e0248e6d/requirements.txt#L5

marcelveldt commented on music_assistant/server/providers/plex/__init__.py:
```diff
@@ -179,6 +188,22 @@ async def get_config_entries(
+            label="Verify certificate",
+            description="Verify local server SSL certificate",
+            required=True,
+            value=values.get(CONF_LOCAL_SERVER_VERIFY_CERT) if values else None,
```
```suggestion
            default_value=True,
```

tenfire replied:
Updated the PR with this suggestion

marcelveldt commented on music_assistant/server/providers/plex/__init__.py:
```diff
@@ -179,6 +188,22 @@ async def get_config_entries(
+            label="SSL (HTTPS)",
+            description="Connect to the local server using SSL (HTTPS)",
+            required=True,
+            value=values.get(CONF_LOCAL_SERVER_SSL) if values else None,
```
```suggestion
            default_value=False,
```

marcelveldt commented:
Can you have a look at the linting errors please ?

tenfire commented:
> Can you have a look at the linting errors please ?

I have fixed the trailing whitespaces, but I do not understand the import sorting rules. Where can I move "import requests"?

marcelveldt reviewed (changes requested):
some more linting required

tenfire commented:
> > > Can you have a look at the linting errors please ?
> > 
> > 
> > I have fixed the trailing whitespaces, but I do not understand the import sorting rules. Where can I move "import requests"?
> 
> Let the linters do that for you:
> 
> in your development venv do "pip install -e ".[test]" to install all test/lint requirements. After that you can do "pre-commit install" to install the git hooks to run the linters

Thank you, I think I have fixed the linting issues now.

marcelveldt reviewed (approved):
Thanks!
